File: atcoder/abc178B_product_max.hs
-- uninhm
-- https://atcoder.jp/contests/abc178/tasks/abc178_b
-- math

main = do
  li <- getLine
  let a:b:c:d:[] = map read $ words li :: [Integer]
  print $ foldl max (-10000000000000000001) [a * c, a * d, b * c, b * d]

File: atcoder/abc178B_product_max.lua
-- vicfred
-- https://atcoder.jp/contests/abc178/tasks/abc178_b
-- math

a, b, c, d = io.read("*n", "*n", "*n", "*n")

maxima = c*b
maxima = math.max(maxima, d*b)
maxima = math.max(maxima, c*a)
maxima = math.max(maxima, a*d)

print(maxima)

